### ISSUE TITLE
Random FreeSolv subset issues

### ISSUE THREAD

**Issue by jchodera** [choderalab/gbff#20]
It looks like the `--subset` flag causes a random subset to be selected:
https://github.com/choderalab/gbff/blob/master/parameterize-using-database.py#L110-L117

```
    # DEBUG: Create a small subset. Do this randomly
    if options.subset_size:
        subset_size = options.subset_size
        cid_list = database.keys()
        max_num = len(cid_list)
        mol_indices = np.random.choice(max_num, subset_size)
        mols_to_use = [cid_list[k] for k in mol_indices]
        database = dict((k, database[k]) for k in mols_to_use)
```

We need to analyze the performance on both the training and test sets.  Is there some way we can easily reconstruct the list of molecules that were used in the training set so we can separate performance by training and test sets?

I wonder if the easiest way to do this is to actually include _all_ molecules in the `database` here, but only include a subset of the molecules (the training set) in the likelihood function.  

We also want to make sure we store
- the list of molecules in the training set
- the list of molecules in the test set
- the DeltaG for each compound, so we can analyze errors by compound class; I'm not sure if these are currently being stored---the previous version of the code defined them as deterministics so that they would be stored in the pymc database
- the RMSE for the training set each iteration; this could be a deterministic, even if it is unused
- the RMSE for the test set each iteration; this could be a deterministic, even if it is unused

**Comment by pgrinaway** [comment]
> Is there some way we can easily reconstruct the list of molecules 

Yes, I had a couple lines in my local clone that wrote out the list of `cid`s

> the DeltaG for each compound

Stored in `db.dg_gbsa()`

> the RMSE for the training set each iteration

Stored in `db.RMSE()`

> the RMSE for the test set each iteration; this could be a deterministic

Good point!

Also, I should point out that `300_adaptive_3gbmodel.h5`, despite having fewer samples, has the same approximate number of effective samples and includes sampling between HCT, OBC, and OBC2.
